src/Compiler/Microsoft.AspNetCore.Razor.Language/test/TestFiles/IntegrationTests/ComponentDesignTimeCodeGenerationTest/AddAttribute_ImplicitStringConversion_Bind/TestComponent.codegen.cs

```diff
@@ -0,0 +1,140 @@
+#line hidden
+#nullable disable
+            , -1, "", -1, 
+#nullable restore
+#line 4 "x:\dir\subdir\Test\TestComponent.cshtml"
+                       () => { }
+
+#line default
+#line hidden
+#nullable disable
+            , -1, 
+#nullable restore
+#line 5 "x:\dir\subdir\Test\TestComponent.cshtml"
+                     c
+
+#line default
+#line hidden
+#nullable disable
+            , -1, 
+#nullable restore
+#line 1 "x:\dir\subdir\Test\TestComponent.cshtml"
+                                c
+
+#line default
+#line hidden
+#nullable disable
+            , -1, global::Microsoft.AspNetCore.Components.EventCallback.Factory.Create(this, 
+            global::Microsoft.AspNetCore.Components.CompilerServices.RuntimeHelpers.CreateInferredEventCallback(this, __value => c = __value, c)));
+            __o = __typeInference_CreateMyComponent_0.
+#nullable restore
+#line 2 "x:\dir\subdir\Test\TestComponent.cshtml"
+    BoolParameter
+
+#line default
+#line hidden
+#nullable disable
+            ;
+            __o = __typeInference_CreateMyComponent_0.
+#nullable restore
+#line 3 "x:\dir\subdir\Test\TestComponent.cshtml"
+    StringParameter
+
+#line default
+#line hidden
+#nullable disable
+            ;
+            __o = __typeInference_CreateMyComponent_0.
+#nullable restore
+#line 4 "x:\dir\subdir\Test\TestComponent.cshtml"
+    DelegateParameter
+
+#line default
+#line hidden
+#nullable disable
+            ;
+            __o = __typeInference_CreateMyComponent_0.
+#nullable restore
+#line 5 "x:\dir\subdir\Test\TestComponent.cshtml"
+    ObjectParameter
+
+#line default
+#line hidden
+#nullable disable
+            ;
+            __o = __typeInference_CreateMyComponent_0.
+#nullable restore
+#line 1 "x:\dir\subdir\Test\TestComponent.cshtml"
+                   MyParameter
+
+#line default
+#line hidden
+#nullable disable
+            ;
+#nullable restore
+#line 1 "x:\dir\subdir\Test\TestComponent.cshtml"
+__o = typeof(global::Test.MyComponent<>);
+
+#line default
+#line hidden
+#nullable disable
+        }
+        #pragma warning restore 1998
+#nullable restore
+#line 7 "x:\dir\subdir\Test\TestComponent.cshtml"
+       
+    private MyClass<string> c = new();
+
+#line default
+#line hidden
+#nullable disable
+    }
+}
+namespace __Blazor.Test.TestComponent
+{
+    #line hidden
+    internal static class TypeInference
+    {
+        public static global::Test.MyComponent<T> CreateMyComponent_0<T>(global::Microsoft.AspNetCore.Components.Rendering.RenderTreeBuilder __builder, int seq, int __seq0, global::System.Boolean __arg0, int __seq1, global::System.String __arg1, int __seq2, global::System.Delegate __arg2, int __seq3, global::System.Object __arg3, int __seq4, global::Test.MyClass<T> __arg4, int __seq5, global::Microsoft.AspNetCore.Components.EventCallback<global::Test.MyClass<T>> __arg5)
+        {
+        __builder.OpenComponent<global::Test.MyComponent<T>>(seq);
+        __builder.AddAttribute(__seq0, "BoolParameter", __arg0);
```

Comment:
> Whats the reason for special casing bool in general?

Same question from me. Keeping the old behavior for `bool` parameters means the bug still exists if the parameter value's type defines an implicit conversion operator to `bool`.

Reply:
> Whats the reason for special casing `bool` in general?

That was meant as an optimization, because `AddAttribute(..., bool)` handles boxing of the `bool` more efficiently (it uses pre-computed `BoxedTrue` and `BoxedFalse` constants). Without special handling it, `AddAttribute(..., (object)BoolArg)` gets called - boxing `true` or `false` at the callsite.

> Keeping the old behavior for `bool` parameters means the bug still exists if the parameter value's type defines an implicit conversion operator to `bool`.

That's a good catch, thanks. I will remove the special casing to fix this.